### PR TITLE
Run image build using kaniko credential

### PR DESCRIPTION
**Pull request by pradithya**
**What this PR does / why we need it**:

`gsutil` command within the pyfunc and batch predictor docker image build failed with GCS access error when running on workload identity node pool even though kaniko pod has been given admin privilege to GCS. Apparently, the gsutil credential is not properly initialize and run without using the kaniko's credential.  This PR properly initialize gsutil credential to use the one provided to the kaniko pod. (similar to https://github.com/caraml-dev/turing/pull/291)

**Does this PR introduce a user-facing change?**:

None

```release-note

```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [x] Tested locally

### CODE REVIEW

File: python/batch-predictor/docker/app.Dockerfile [hunk 1]
```diff
@@ -18,8 +18,12 @@ FROM ${BASE_IMAGE}
 # Get the model
 
 ARG MODEL_URL
+ARG GOOGLE_APPLICATION_CREDENTIALS
+
+# Run docker build using the preovided credential
+RUN gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
```
Comment (pradithya) [inline python/batch-predictor/docker/app.Dockerfile:24]: @krithika369 I wonder if the service account file will already be included in the docker build context?

Comment (krithika369) [reply python/batch-predictor/docker/app.Dockerfile:24]: That's a good question. Tested it e2e for Turing and it works. So it's available.

File: python/batch-predictor/docker/app.Dockerfile [hunk 2]
```diff
@@ -18,8 +18,12 @@ FROM ${BASE_IMAGE}
 # Get the model
 
 ARG MODEL_URL
+ARG GOOGLE_APPLICATION_CREDENTIALS
+
+# Run docker build using the preovided credential
```
Comment (tiopramayudi) [inline python/batch-predictor/docker/app.Dockerfile:23]: nit: typo here

**Comment by tiopramayudi** [comment]
LGTM

**Review by tiopramayudi — APPROVED** [review]

**Review by ariefrahmansyah — APPROVED** [review]

**Pull request merged**